app/main.py

```diff
@@ -1,8 +1,28 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+        Person.people[name] = self
+
+
+def create_person_list(people_list: list) -> list:
+    new_people_list = [Person(person["name"], person["age"])
+                       for person in people_list]
+
+    for person in people_list:
+        if "husband" in person and person["husband"]:
+            for woman in new_people_list:
+                for man in new_people_list:
+                    if (woman.name == person["name"]
+                            and man.name == person["husband"]):
+                        woman.husband = man
```

Comment:
You could avoid the double loop iteration by changing the logic of the check. You could only check if there is a `"husband"` key and in case it is present, you could extract the person instance that is currently used in iteration, and extract the husband person instance by husband name from the `Person.people` class attribute. After that you could create the necessary `husband` and `wife` attributes in both instances. The same logic could be applied in case `"wife"` key is present.